### PR TITLE
Comment edit

### PR DESCRIPTION
**Pull request by bradleypickard**
Not pretty, but enables comment editing on comments the user has created

### CODE REVIEW

**Comment by hari387** [comment]
You may as well finish up the server side auth for comment PUT using `Comment.createdByUser` now that you've got my auth stuff here.

File: src/main/webapp/recipe-display.html [hunk 1]
```diff
@@ -135,9 +136,11 @@
           })
       }
 
+      /* Deletes the currently displayed post and all associated comments.
+       *   TODO: Only allow this operation to be taken by post creator
```
Comment (nonsensicle) [inline src/main/webapp/recipe-display.html:140]: This is taken care of on the back end, but yeah ideally we'll have the delete/edit posts and edit comment html elements visible only to the creator

File: src/main/webapp/recipe-display.html [hunk 2]
```diff
@@ -149,15 +152,92 @@
...
+                  "&token=" +
+                  idToken,
+                {
+                  method: "PUT",
+                  headers: {
+                    "Content-Type": "application/json",
+                  },
+                }
+              ).then((response) => {
+                if (response.status < 200 || response.status >= 300) {
```
Comment (nonsensicle) [inline src/main/webapp/recipe-display.html:204]: Use `!response.ok` for consistency

File: src/main/webapp/recipe-display.html [hunk 3]
```diff
@@ -149,15 +152,92 @@
...
+
+        editForm.value = commentEl.querySelector("p").innerText
+        submitEdit.innerText = "Submit"
+        submitEdit.addEventListener("click", submitEditEvent)
+      }
+
+      /* Event triggered by clicking Submit Button under Edit
+       * Removes edit form, calls function to upload changes to DB
+       */
+      function submitEditEvent(e) {
```
Comment (nonsensicle) [inline src/main/webapp/recipe-display.html:172]: Maybe should add a "cancel" button to cancel the action and then refactor this function to switch between cancel and submit logic? Not necessary but could improve the user experience!

**Pull request merged**